include/flash_map.h

```diff
@@ -33,8 +33,13 @@ extern "C" {
 #include <stddef.h>
 #include <sys/types.h>
 
-#define SOC_FLASH_0_ID 0	/* device_id for SoC flash memory driver */
-#define SPI_FLASH_0_ID 1	/* device_id for external SPI flash driver */
+/**
+ * @brief Flash Area
+ * @defgroup flashmap_api Flah Area
```

Comment:
Fix typo, `Flah' -> `Flash`.  Also do you mind marking this as deprecated.

Reply:
what should be deprecated?